src/app/pages/things/details/things.details.component.ts

```diff
@@ -135,6 +146,37 @@ export class ThingsDetailsComponent implements OnInit {
     }
   }
 
+  onChangeLimitDisc(limit: number) {
+    this.findDisconnectedChans(0, limit);
+  }
+
+  onChangePageDisc(dir: any) {
+    if (dir === 'prev') {
+      const offset = this.disconnChansPage.offset - this.disconnChansPage.limit;
+      this.findDisconnectedChans(offset, this.disconnChansPage.limit);
+    }
+    if (dir === 'next') {
+      const offset = this.disconnChansPage.offset + this.disconnChansPage.limit;
+      this.findDisconnectedChans(offset, this.disconnChansPage.limit);
+    }
+  }
+
+  onCheckboxConns(row: any) {
+    if (this.selectedToConn.includes(row.id)) {
+      this.selectedToConn = this.selectedToConn.filter(id => id !== row.id);
+    } else {
+      this.selectedToConn.push(row.id);
+    }
+  }
+
+  onCheckboxDisconns(row: any) {
+    if (this.selectedToDisconn.includes(row.id)) {
+      this.selectedToDisconn = this.selectedToDisconn.filter(id => id !== row.id);
+    } else {
+      this.selectedToDisconn.push(row.id);
+    }
+  }
+
```

Comment:
You can avoid one of these tests by checking the length of the `this.selectedToConn.filter(id => id !== row.id)` return value. `filter` returns an empty array if there is nothing filtered. So you could do something like 

```
this.selectedToDisconn = this.selectedToDisconn.filter(id => id !== row.id)
if (!this.selectedToDisconn.length) this.selectedToDisconn.push(row.id);
```
It's much cheaper to test for length than whether array includes something.

Reply:
What happens if there is already an ID in `selectedToDisconn` ?

Reply:
The length of the array will be greater than zero.